File: Assets/ToLua/Lua/user/ui/vip/controller/vip_controller.lua
local t = {}
local name = 'vip_controller'
require 'vip_pb'

local vipModel = gamemanager.GetModel('vip_model')

t.lastDrawBenefitsVIPLevel = -1

local function Start ()
  netmanager.RegisterProtocol(MSG.VipInfoResp, t.VipInfoResp)
  netmanager.RegisterProtocol(MSG.VipGiftBagResp, t.VipGiftBagResp)
  
  gamemanager.RegisterCtrl(name,t)
end

function t.VipInfoReq ()
  local req = vip_pb.VIPInfoReq()
  netmanager.SendProtocol(MSG.VIPInfoReq, req)
end

function t.VipInfoResp ()
  print('========== VipInfoResp ==========')
  local resp = vip_pb.VipInfoResp()
  resp:ParseFromString(netmanager.GetProtocolData())
  
  vipModel.UpdateVIPInfo(resp.dailyGiftIsGet, resp.totalRecharge, resp.getGiftBagLvList)
end

function t.VipGiftBagReq (vipLevel)
  local req = vip_pb.VipGiftBagReq()
  req.vipLv = vipLevel
  t.lastDrawBenefitsVIPLevel = vipLevel
  netmanager.SendProtocol(MSG.VipGiftBagReq, req)
end

function t.VipGiftBagResp ()
  print('========== VipInfoResp ==========')
  local resp = vip_pb.VipGiftBagResp()
  resp:ParseFromString(netmanager.GetProtocolData())
  vipModel.DrawVIPBenefitsSuccess(t.lastDrawBenefitsVIPLevel)
end

function t.OpenVIPView ()
  local vip_view = require('ui/vip/view/vip_view')
  vip_view.Open()
end

Start ()
return t

File: Assets/Code/Logic/VIP/Controller/VIPController.cs
using UnityEngine;
using Logic.Protocol.Model;
using Logic.VIP.Model;

namespace Logic.VIP.Controller
{
	public class VIPController : SingletonMono<VIPController>
	{
		private int _lastDrawBenefitsVIPLevel = -1;

		void Awake ()
		{
			instance = this;
		}

		/*
		void Start ()
		{
			Observers.Facade.Instance.RegisterObserver(((int)MSG.VipInfoResp).ToString(), S2C_VIPInfoResp);
			Observers.Facade.Instance.RegisterObserver(((int)MSG.VipGiftBagResp).ToString(), S2C_VIPGiftBagResp);
		}

		void OnDestroy ()
		{
			if (Observers.Facade.Instance != null)
			{
				Observers.Facade.Instance.RemoveObserver(((int)MSG.VipInfoResp).ToString(), S2C_VIPInfoResp);
				Observers.Facade.Instance.RemoveObserver(((int)MSG.VipGiftBagResp).ToString(), S2C_VIPGiftBagResp);
			}
		}

		public void C2S_VIPInfoReq ()
		{
			VipInfoReq vipInfoReq = new VipInfoReq ();
			Logic.Protocol.ProtocolProxy.instance.SendProtocol(vipInfoReq);
		}

		public bool S2C_VIPInfoResp (Observers.Interfaces.INotification note)
		{
			VipInfoResp vipInfoResp = note.Body as VipInfoResp;
			VIPProxy.instance.UpdateVIPInfo(vipInfoResp.dailyGiftIsGet,vipInfoResp.totalRecharge, vipInfoResp.getGiftBagLvList);
			return true;
		}

		public void C2S_VIPGiftBagReq (int vipLevel)
		{
			VipGiftBagReq vipGiftBagReq = new VipGiftBagReq();
			vipGiftBagReq.vipLv = vipLevel;
			_lastDrawBenefitsVIPLevel = vipLevel;
			Logic.Protocol.ProtocolProxy.instance.SendProtocol(vipGiftBagReq);
		}

		public bool S2C_VIPGiftBagResp (Observers.Interfaces.INotification note)
		{
            LuaInterface.LuaScriptMgr.Instance.CallLuaFunctionVoid("ActivityRspVipGiftBag");
            VipGiftBagResp vipGiftBagResp = note.Body as VipGiftBagResp;
            if (VIPProxy.instance != null)
                VIPProxy.instance.onDrawVIPBenefitsSuccessDelegate(_lastDrawBenefitsVIPLevel);
			return true;
		}
		*/
	}
}
